packages/cli/create-strapi-starter/utils/build-starter.js

```diff
@@ -86,29 +87,52 @@ async function installWithLogs(path) {
 }
 
 /**
- * @param  {Object} projectArgs - The arguments for create a project
- * @param {string|null} projectArgs.projectName - The name/path of project
- * @param {string|null} projectArgs.starterUrl - The GitHub repo of the starter
- * @param  {Object} program - Commands for generating new application
+ * @param {string} starter The name of the starter as provided by the user
+ * @param {boolean} useYarn Use yarn instead of npm
  */
-module.exports = async function buildStarter(programArgs, program) {
-  let { projectName, starterUrl } = programArgs;
-
-  // Fetch repo info
-  const repoInfo = await getRepoInfo(starterUrl);
-  const { fullName } = repoInfo;
-
-  // Create temporary directory for starter
-  const tmpDir = await fse.mkdtemp(join(os.tmpdir(), 'strapi-'));
+async function getStarterInfo(starter, useYarn) {
```

Comment:
you should pass useYarn in an options object to make this function a bit more flexible